src/pgduckdb.cpp

```diff
@@ -95,6 +96,10 @@ DuckdbInitGUC(void) {
 
 	DefineCustomVariable("duckdb.max_memory", "The maximum memory DuckDB can use (e.g., 1GB)", &duckdb_maximum_memory);
 
+	DefineCustomVariable("duckdb.disabled_filesystems",
+	                     "Disable specific file systems preventing access (e.g., LocalFileSystem)",
+	                     &duckdb_disabled_filesystems);
```

Comment:
Needs to be PGC_SUSET

Reply:
That's a good point. And that makes me think that these settings won't be taken into account if a (super)user try to update them at runtime unless they call `reset_ddb` method. Probably not a big deal for now but we might want to track it WDYT?

Reply:
Yeah we should probably create an issue to track that, similar to how we have #278. Below some thoughts on the implementation:

For settings that DuckDB allows changing on the fly using `SET` commands this should be relatively easy/elegant to implement. We can use the GucAssignHook for each of these GUCs to run a SET on the database (if it is initialized), whenever the GUC is changed.

For GUCs that are not possible to change without resetting the DuckDB database (such as `threads` afaik), I think there are two options:
1. We change PGC_SUSET to PGC_SUBACKEND, so that changing these settings after connecting to postgres is disallowed.
2. We have the assignhook log a WARNING/NOTICE (if the DuckDB db is initialized) that the change will only be picked up after calling `reset_ddb`.

I think the choice there is how much we want to advertise `reset_ddb`.